lib/dhcp/client.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dhcp
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"syscall"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/d2g/dhcp4"
+	"github.com/d2g/dhcp4client"
+)
+
+// Client represents a DHCP client
+type Client interface {
+	// SetTimeout sets the timeout for a subsequent DHCP request
+	SetTimeout(t time.Duration) error
+
+	// Request sends a full DHCP request, resulting in a DHCP lease.
+	// On a successful lease, returns a DHCP acknowledgment packet
+	Request(int, net.HardwareAddr) (*Packet, error)
+
+	// Renew renews an existing DHCP lease. Returns a new acknowledgment
+	// packet on success.
+	Renew(*Packet) (*Packet, error)
+
+	// Release releases an existing DHCP lease.
+	Release(*Packet) error
+}
+
+type client struct {
+	sync.Mutex
+
+	timeout time.Duration
+}
+
+// The default timeout for the client
+const defaultTimeout = 10 * time.Second
+
+// NewClient creates a new DHCP client. Note the returned object is not thread-safe.
+func NewClient() (Client, error) {
+	return &client{timeout: defaultTimeout}, nil
+}
+
+func (c *client) SetTimeout(t time.Duration) error {
```

Comment:
Why we are returning error here? I think we can just drop that and return nothing